File: lookuptable_pullers/traildiscover/traildiscover_puller.py
import json
from pathlib import Path

import jsonlines

# Specify the directory containing the events and the output file path
events_directory = "./events"
output_file = "./docs/traildiscover.jsonl"
pathlist = Path(events_directory).rglob("*.json")

# Initialize an empty list to store all events
compiled_events = []
event_names = set()

# Read all events and store them in the list
for file in pathlist:
    with open(file, "r") as f:
        event_data = json.load(f)
    # Ensure that event names are unique
    if event_data["eventName"] not in event_names:
        event_names.add(event_data["eventName"])
        compiled_events.append(event_data)

# Write the compiled events to the output file
with jsonlines.open(output_file, mode="w") as writer:
    writer.write_all(compiled_events)

print(f"Compiled events into '{output_file}' successfully.")

File: lookuptable_pullers/traildiscover/traildiscover_puller.go
package main

import (
	"bytes"
	"encoding/json"
	"io"
	"os"
	"path/filepath"
	"slices"
)

type Event struct {
	EventName string `json:"eventName"`
}

func main() {

	var eventsDirectory = "./events"
	var outputFileDirectory = "./docs/traildiscover.jsonl"

	fileList := make([]string, 0, 10)
	eventNamesList := make([]string, 0, 10)

	_ = filepath.Walk(eventsDirectory, func(path string, info os.FileInfo, err error) error {
		if filepath.Ext(path) == ".json" {
			fileList = append(fileList, path)
		}
		return nil
	})

	outputFile, _ := os.OpenFile(outputFileDirectory, os.O_APPEND|os.O_CREATE|os.O_WRONLY, os.ModePerm)

	for _, file := range fileList {
		jsonFile, _ := os.Open(file)
		defer jsonFile.Close()

		byteValue, _ := io.ReadAll(jsonFile)

		var event Event
		_ = json.Unmarshal(byteValue, &event)

		buffer := new(bytes.Buffer)
		_ = json.Compact(buffer, byteValue)
		trimmedByteValue := buffer.Bytes()

		if !slices.Contains(eventNamesList, event.EventName) {
			eventNamesList = append(eventNamesList, event.EventName)

			outputFile.Write(trimmedByteValue)
			outputFile.Write([]byte("\n"))
		}
	}
	println("Compiled events into", outputFileDirectory, "successfully.")
	defer outputFile.Close()
}
